examples/allocation-endpoint/README.md

```diff
@@ -0,0 +1,103 @@
+
+```
+kubectl patch deployment agones-allocator -n agones-system --patch-file patch-agones-allocator.yaml
+kubectl patch svc agones-allocator -n agones-system --type merge -p '{"spec":{"ports": [{"port": 443,"name":"https","targetPort":9443}]}}'
+kubectl annotate sa -n agones-system agones-allocator iam.gke.io/gcp-service-account=ae-esp-sa@[PROJECT-ID].iam.gserviceaccount.com
+```
+
+## Terraform 
+The terraform modules create resources in GCP:
+
+```
+terraform apply \
+  -var "project_id=[PROJECT-ID]" \
+  -var "authorized_members=[\"serviceAccount:[SERVICE-ACCOUNT-EMAIL]\"]" \
+  -var "clusters_info=[CLUSTERS-INFO]" \
+  -var "workload-pool=[GKE-PROJECT-ID].svc.id.goog"
+```
+
+`[CLUSTERS-INFO]` is in the form of `[{\"name\":\"cluster1\",\"endpoint\":\"34.83.14.82\",\"namespace\":\"default\",\"allocation_weight\":100},{...}]` deserializing to []ClusterInfo, defined in the `server/clusterselector.go`.
+
+- The `name` is a unique randomly selected name for the cluster.
+- The `endpoint` is the `agones-allocator` external IP.
+- The `namespace` is the game server namespace.
+- The `allocation_weight` is a value between 0 and 100, which sets the relative allocation rate a cluster receives compared to other clusters. By setting weight to zero, a cluster stops receiving allocation requests.
+
+`[SERVICE-ACCOUNT-EMAIL]` is the service account to be granted access the Allocation Endpoint. You need to have [the service account created](https://cloud.google.com/iam/docs/creating-managing-service-accounts) before running terraform.
+
+## Server
+
+The Allocation Endpoint proxy code is in `./server` folder. You can make changes and run the following to build and push the image to your own GCR repository:
+
+```
+docker build --tag gcr.io/[PROJECT-ID]/allocation-endpoint-proxy:[VERSION] .
+docker push gcr.io/[PROJECT-ID]/allocation-endpoint-proxy:[VERSION]
+```
+
+If you are building your own image, you can set `ae_proxy_image` terraform variable to your image.
+
+## Client
+
+The Allocation Endpoint client code is in `./client` folder. Get the Service Account Key for one of the Service Accounts in the list of `authorized_members` and put it under `sa_key.json`. Alternatively, you can leverage [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) to retrieve the access token using default service account from metadata serer, when deploying your client in a GCP solution, e.g. GKE.
+
+```
+go run *.go --url=[CLOUD-RUN-ENDPOINT]
+
+```
+
+`[CLOUD-RUN-ENDPOINT]` is the cloud run endpoint FQDN printed out after running the terraform. Leave out the scheme when setting the value e.g. `allocation-endpoint-proxy-<code>.a.run.app`.
+
+## Future considerations
+- Requests using this example goes to public IP. For clusters in the same project you can instead leverage VPC with private IPs and remove dependency to the Service Account and Secret Manager to issue JWT in the proxy.
```

Comment:
*(non blocking) Question:* - Do we think this work with [Internal Only Load Balancers](https://cloud.google.com/kubernetes-engine/docs/how-to/internal-load-balancing)? Rather than a whole VPC configuration?

Reply:
Yes, cloud run can form a [VPC access](https://cloud.google.com/vpc/docs/configure-serverless-vpc-access). It is a 3 step setup, (1) create an internal LB for the server; (2) create a VPC access connector and (3) configure cloud run to use the VPC connector.

Reply:
TIL. Thanks!